libraries/integer/java/Utils.java

```diff
@@ -59,6 +59,10 @@ private static BigInteger toUnsignedBigInteger(long i) {
         }
     }
 
+    public static BigInteger toUnsignedBigInteger(int i) {
+      return Utils.toUnsignedBigInteger(Long.valueOf(i));
```

Comment:
Ahh i originally put that, but was getting a weird result, since `-1` ends up padded then converted it ends up as
```
-1 (input)
0x00000000ffffffff (output)
```
where as passing the long directly in to get converted i get
```
-1 (input)
0xffffffffffffffff (output)
```
I guess the actual problem is a 32bit integer is being used to represent a pointer on a 64bit platform so there is data loss in the conversion? I'll switch it over to use `toUnsignedLong` when i get home